src/interfaces/nav-controller-interface.ts

```diff
@@ -0,0 +1,31 @@
+import { LitElement } from 'lit-element';
+import { MetadataResponse } from '@internetarchive/search-service';
+import { ModalManager } from '@internetarchive/modal-manager';
+import { SharedResizeObserver } from '@internetarchive/shared-resize-observer';
+import { IntMenuShortcut } from './menu-interfaces';
+
+export interface IntNavController extends LitElement {
+  baseHost: string;
+  itemMD: MetadataResponse;
+  menuProviders: object;
+  menuShortcuts: IntMenuShortcut[];
+  sideMenuOpen: boolean;
+
+  signedIn: boolean;
+
+  sharedObserver: SharedResizeObserver;
+  modal: ModalManager;
+
+  emitLoadingStatusUpdate: (loaded: boolean) => void;
+
+  addMenuShortcut: (menuId: string) => void;
+  removeMenuShortcut: (menuId: string) => void;
+  sortMenuShortcuts: () => void;
+  emitMenuShortcutsUpdated: () => void;
+
+  book: MetadataResponse;
+}
+
+export interface BookNavigator extends IntNavController {
```

Comment:
Q: How come you have an interface `BookNavigator` and a class `BookNavigator` below, but the `BN` class doesn't implement the `BN` interface?

Reply:
I might suffix the name of this interface with `Interface`... without it, it looks like it's an implementation. Same with `IntNavController`.